core/src/components/cat-radio-group/cat-radio-group.tsx

```diff
@@ -0,0 +1,76 @@
+import { Component, Host, h, Prop, Element, Listen } from '@stencil/core';
+
+@Component({
+  tag: 'cat-radio-group',
+  styleUrl: 'cat-radio-group.scss',
+  shadow: true
+})
+export class CatRadioGroup {
+  private catRadioGroup: HTMLCatRadioElement[] = [];
+
+  @Element() hostElement!: HTMLElement;
+
+  /**
+   * The name of the radio group component.
+   */
+  @Prop() name?: string;
+
+  /**
+   * Whether this radio group is disabled.
+   */
+  @Prop() disabled = false;
+
+  componentDidLoad(): void {
+    this.catRadioGroup = Array.from(this.hostElement.querySelectorAll(`cat-radio`));
+    this.catRadioGroup.forEach(catRadio => {
```

Comment:
this will not work if the property is changed later. Or toggled dynamically. We need to execute it on watch of the property as well as on load. Probably applies to both the disabled and the name.